### DISCUSSION TITLE
GPU Requirements for yolov5 inference

### DISCUSSION THREAD

**Discussion by dwhyt** [ultralytics/yolov5#6720] [Q&A]
are any minimum GPU requirements for inference using a custom trained yolov5 model? i am using the x version and would like to carry out my detection at 960px, not sure if a 5gb gpu is adequate

**Comment by glenn-jocher** [comment]
@dwhyt you can use the YOLOv5 `profile` function to determine resource utilization by YOLOv5 models:

```python
# Profile
import torch
from utils.torch_utils import profile

model = torch.hub.load('ultralytics/yolov5', 'yolov5s')

results = profile(input=torch.randn(1, 3, 640, 640), ops=[model], n=30)
```

<img width="1232" alt="Screenshot 2022-02-21 at 12 46 30" src="https://user-images.githubusercontent.com/26833433/154949431-63ad5132-c770-43b1-afc8-28828a4761a1.png">
